modules/InboundEmail/InboundEmail.php

```diff
@@ -2673,108 +2676,101 @@ public function savePersonalEmailAccount($userId = '', $userName = '', $forceSav
-                        'is_dynamic' => 1,
-                        'created_by' => $focusUser->id,
-                        'modified_by' => $focusUser->id,
-                    ),
-                );
-
-
-                require_once("include/SugarFolders/SugarFolders.php");
-
-                $parent_id = '';
-
 
-                foreach ($params as $type => $type_params) {
-                    if ($type == "inbound") {
+            if (empty($row)) {
 
-                        $folder = new SugarFolder();
-                        foreach ($params[$type] as $key => $val) {
-                            $folder->$key = $val;
-                        }
-
-                        $folder->new_with_id = false;
-                        $folder->id = $this->id;
-                        $folder->save();
-
-                        $parent_id = $folder->id;
-                    } else {
-                        $params[$type]['parent_folder'] = $parent_id;
-
-                        $folder = new SugarFolder();
-                        foreach ($params[$type] as $key => $val) {
-                            $folder->$key = $val;
-                        }
+                $this->createFolder(
+                    $inboxFolders[0] . ' ('.$this->name.')',
+                    "inbound",
+                    $focusUser,
+                    $this->id
+                );
 
-                        $folder->save();
+                foreach ($inboxFolders as $key => $folder) {
+                    if ($key == 0) {
+                        continue;
+                    }
+                    if ($this->folderIsRequestTrashOrSent($folder)) {
+                        continue;
                     }
+                    $this->createFolder(
+                        $folder,
+                        "inbound",
+                        $focusUser
+                    );
                 }
+
+                $this->createFolder(
+                    $mod_strings['LNK_MY_DRAFTS'] . ' (' . $stored_options['sentFolder'] . ')',
+                    "draft",
+                    $focusUser
+                );
+                $this->createFolder(
+                    $mod_strings['LNK_SENT_EMAIL_LIST'] . ' (' . $stored_options['sentFolder'] . ')',
+                    "sent",
+                    $focusUser
+                );
+                $this->createFolder(
+                    $mod_strings['LBL_LIST_TITLE_MY_ARCHIVES'],
+                    "archived",
+                    $focusUser
+                );
+
             } else {
                 // Update folders
-                $foldersFound = $this->db->query('SELECT * FROM folders WHERE folders.id LIKE "'.$this->id.'" OR '.
-                    'folders.parent_folder LIKE "'.$this->id.'"');
-                while($row = $this->db->fetchRow($foldersFound)) {
+                $foldersFound = $this->db->query(
+                    'SELECT * FROM folders WHERE folders.id LIKE "' . $this->id . '" OR ' .
```

Comment:
You can use $this->db->quoted here.

Reply:
I didn't actually change this line of code, just the indentation.

Reply:
@PedroErnst You right, it's a separated issue but can you please using a `$db->quote()` function to prevent any problem here and above also.